zcash_client_sqlite/src/wallet.rs

```diff
@@ -179,6 +179,60 @@ pub fn get_address<P: consensus::Parameters>(
         })
 }
 
+pub(crate) fn get_max_account_id<P>(
+    wdb: &WalletDb<P>,
+) -> Result<Option<AccountId>, SqliteClientError> {
+    // This returns the most recently generated address.
+    Ok(wdb
+        .conn
+        .query_row("SELECT MAX(account) FROM accounts", NO_PARAMS, |row| {
```

Comment:
We should maybe exclude `u31::MAX` from this selection, just in case the legacy account from `zcashd` somehow ends up in the wallet database (it shouldn't as there is by definition no ZIP 316 account at that position, but databases are fun). This would mean the code continues to correctly add accounts sequentially from whatever is the earliest non-legacy account ID. We would need to separately handle the case where this returns `u31::MAX - 1`, in which case the caller would currently increment that and then either incorrectly add the legacy account ID, or clobber whatever is there.

Reply:
Fixed in https://github.com/zcash/librustzcash/pull/637/commits/55e3afdbcf3a499d1dd59d5c33104c0d4e0e9d2e